apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -160,11 +160,29 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     return dependency.resolution.tarball;
   }
 
+  /**
+   * Gets the dependency key name of the given temporary project
+   * Examples of the return value:
+   *   file:projects/empty-webpart-project.tgz
+   *   file:projects/article-site-demo.tgz_jest@22.4.4+typescript@2.9.2
+   *   file:projects/i18n-utilities.tgz_462eaf34881863298955eb323c130fc7
+   */
+  public getTempProjectKey(tempProjectName: string): string {
```

Comment:
Hmm, it doesn't make sense to expose a method called `getTempProjectKey `from the class `PnpmShrinkWrapFile`. I will rename this.